sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuTextBasedPartitionReader.scala

```diff
@@ -384,6 +387,19 @@ abstract class GpuTextBasedPartitionReader[BUFF <: LineBufferer, FACT <: LineBuf
     }
   }
 
+  private lazy val toUTF8Bytes: Text => (Array[Byte], Int) =
```

Comment:
I know this works, but why we use a lambda here? Is it an utility function?

Reply:
This can avoid executing this `if-else` on each row, to get better performance.